lib/token-assert.js

```diff
@@ -31,19 +31,23 @@ TokenAssert.prototype.whitespaceBetween = function(options) {
         if (nextToken.loc.start.line === token.loc.end.line &&
             (nextToken.loc.start.column - token.loc.end.column) !== spaces
         ) {
+            nextToken.whitespaceBefore = new Array(spaces + 1).join(' ');
```

Comment:
Yes, but this modification should likely only happen if fixing is turned on.

Eventually, rules might actually be checking the whitespaceBefore property, and if we unconditionally change this whitespace, than overlapping rules will fail to report an error at the same location if we're not in fix mode.